airbytehq/airbyte
# connectors-ci: make java connector not use the global dockerd

## What
Addresses #27681 
Java connectors build require an active dockerd daemon to be built because their dependencies use the `airbyteDocker` plugin. Nothing [but this `docker` call ](https://github.com/airbytehq/airbyte/blob/master/buildSrc/src/main/groovy/airbyte-docker.gradle#L231 )would be run if the connector does not have a dockerfile (as it should be the case inside the dagger container).

## Why is this a problem?
The reliance on the global docker daemon that we've have to use in the Dagger context has proven to be a SPOF. When its under heavy use (eg nightly build) the global docker daemon can fail leading to cascading failures.

## Solution
-  Make gradle tasks use their own dockerd
-  Exclude the java connector dockerfile from the container in which we run `distTar` to make minimum usage of the dockerd required by `aircmd`

**We'd like to remove airbyteDocker gradle plugin soon but this change helps mitigate the dockerd reliance on build without changes to grade files**

## Manual testing

* 🟢 Run test pipeline on a couple of java connectors: make sure the build and integration test work (see bot comments)
* 🟢  Run test pipeline on a python connector to check that no regression was introduced (see bot comments)
* 🟢 Run a publish pipeline [on a java](https://github.com/airbytehq/airbyte/actions/runs/5810556131/job/15751725598) and  🟢 [python connectors](https://github.com/airbytehq/airbyte/actions/runs/5810516957/job/15751590426) to check that no regression was introduced
